plugins/spacelift/api_key.go

```diff
@@ -0,0 +1,108 @@
+package spacelift
+
+import (
+	"context"
+
+	"github.com/1Password/shell-plugins/sdk"
+	"github.com/1Password/shell-plugins/sdk/importer"
+	"github.com/1Password/shell-plugins/sdk/provision"
+	"github.com/1Password/shell-plugins/sdk/schema"
+	"github.com/1Password/shell-plugins/sdk/schema/credname"
+	"github.com/1Password/shell-plugins/sdk/schema/fieldname"
+)
+
+func APIKey() schema.CredentialType {
+	return schema.CredentialType{
+		Name:          credname.APIKey,
+		DocsURL:       sdk.URL("https://docs.spacelift.io/integrations/api#spacelift-api-key-token"),
+		ManagementURL: sdk.URL("https://mycorp.app.spacelift.io/settings/api-keys"),
+		Fields: []schema.CredentialField{
+			{
+				Name:                fieldname.Endpoint,
+				MarkdownDescription: "The URL to your Spacelift account, for example https://mycorp.app.spacelift.io",
+				Composition: &schema.ValueComposition{
+					Charset: schema.Charset{
+						Lowercase: true,
+						Digits:    true,
+						Symbols:   true,
+					},
+					Prefix: "https://",
+				},
+			},
+			{
+				Name:                fieldname.APIKeyID,
+				MarkdownDescription: "The ID of your Spacelift API key. Available via the Spacelift application.",
+				Secret:              true,
```

Comment:
I'd expect only the `API_KEY_SECRET` to be secret. Does Spacelift have any documentation on whether the API Key ID should be considered secret?

Reply:
<img width="658" alt="Screenshot 2024-02-27 at 12 36 29" src="https://github.com/1Password/shell-plugins/assets/45081667/6e473ae1-b538-43f9-9f47-b1ce16812063">

From [their docs](https://docs.spacelift.io/concepts/spacectl#usage-on-github-actions) it seems their official GitHub action treats the key id as a secret as well.